File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>arr;
        int st=0;
        int end=0;
        vector<int>one;
        one.push_back(1);
        arr.push_back(one);
        for(int i=1;i<numRows;i++){
            vector<int>two;
            two.push_back(st+arr[i-1][0]);
            for(int j=0;j<arr[i-1].size()-1;j++){
                two.push_back(arr[i-1][j]+arr[i-1][j+1]);
            }
            two.push_back(arr[i-1][arr[i-1].size()-1]+end);
            arr.push_back(two);
        }
        return arr;
    }
};

File: 118-pascals-triangle/118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
         var arr=[];
        var st=0;
        var end=0;
        // vector<int>one;
    var one=[];
        one.push(1);
        arr.push(one);
        for(var i=1;i<numRows;i++){
            var two=[] ;
            two.push(st+arr[i-1][0]);
            for(var j=0;j<arr[i-1].length-1;j++){
                two.push(arr[i-1][j]+arr[i-1][j+1]);
            }
            two.push(arr[i-1][arr[i-1].length-1]+end);
            arr.push(two);
        }
        return arr;
};